src/Model/Sneak.re

```diff
@@ -0,0 +1,118 @@
+open Revery.Math;
+
+type callback = unit => unit;
+type bounds = unit => option(Rectangle.t);
+
+type sneakInfo = {
+  callback,
+  boundingBox: BoundingBox2d.t,
+};
+
+type sneak = {
+  callback,
+  boundingBox: BoundingBox2d.t,
+  id: string,
+};
+
+[@deriving show({with_path: false})]
+type action =
+  | Initiated
+  | Discover([@opaque] list(sneakInfo))
+  | KeyboardInput(string)
+  | Stopped;
+
+type t = {
+  active: bool,
+  allSneaks: list(sneak),
+  prefix: string,
+  filteredSneaks: list(sneak),
+};
+
+let initial: t = {
+  active: false,
+  allSneaks: [],
+  prefix: "",
+  filteredSneaks: [],
+};
+
+let getTextHighlight = (text: string, model: t) => {
+  let prefixLength = model.prefix |> String.length;
+  let idLength = text |> String.length;
+
+  let remainder = String.sub(text, prefixLength, idLength - prefixLength);
+
+  (model.prefix, remainder);
+};
+
+let reset = _sneak => {...initial, active: true};
+
+let hide = _sneak => initial;
+
+let isActive = sneaks => sneaks.active;
+
+let _filter = (prefix: string, sneak: sneak) => {
```

Comment:
I'm partial to using underscore to denote internal functions too, but there are some unintentional issues with it. One ise that syntax highlighting tends to grey them out, making them hard to visually scan for. Another problem is that the compiler uses a leading underscore as a hint that it is intentionally unused and won't give you any warning about it. That might eventually lead to a lot of unused code laying around and cluttering the place up. It can also lead to subtle errors with shadowing, although that's much less likely to occur with top-level definitions.

In any case, I'm not sure it's worth it with these issues. Or perhaps a different scheme would be better, like an `Internal` submodule.

Reply:
I like the idea of an `Internal` submodule 👍